tools/gen-postmortem-metadata.py

```diff
@@ -0,0 +1,169 @@
+    self.name = name
+    self.value = value
+    self.headers = headers
+    self.type_ = type_ or DebugSymbol.type_
+
+  @classmethod
+  def get_headers(cls, debug_symbols):
+    '''
+    Return a list of headers without duplicates, preserving the order they were
+    declared
+    '''
+    seen = set()
+    headers = [debug_symbol.headers for debug_symbol in debug_symbols]
+    headers = sum(headers, [])
+
+    result = []
+    for h in headers:
+      if not h in seen:
+        seen.add(h)
+        result.append(h)
+
+    return result
+
+  @property
+  def declare(self):
+    return '{type} {prefix}{name};'.format(
+      type=self.type_,
+      prefix=self._prefix,
+      name=self.name,
+    )
+
+  @property
+  def fill(self):
+    return '{prefix}{name} = {value};'.format(
+      prefix=self._prefix,
+      name=self.name,
+      value=self.value,
+    )
+
+
+debug_symbols = [
+  DebugSymbol(
+    name='environment_context_idx_embedder_data',
+    value='Environment::kContextEmbedderDataIndex',
+    headers=['env.h'],
+    type_='int',
+  ),
+  DebugSymbol(
+    name='class__BaseObject__persistent_handle',
+    value='offsetof(BaseObject, persistent_handle_)',
+    headers=['base-object.h', 'base-object-inl.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__Environment__handleWrapQueue',
+    value='offsetof(Environment, handle_wrap_queue_)',
+    headers=['env.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__HandleWrap__node',
+    value='offsetof(HandleWrap, handle_wrap_queue_)',
+    headers=['handle_wrap.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__HandleWrapQueue__headOffset',
+    value='offsetof(Environment::HandleWrapQueue, head_)',
+    headers=['env.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__HandleWrapQueue__nextOffset',
+    value='offsetof(ListNode<HandleWrap>, next_)',
+    headers=['handle_wrap.h', 'util.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__Environment__reqWrapQueue',
+    value='offsetof(Environment, req_wrap_queue_)',
+    headers=['env.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__ReqWrap__node',
+    value='offsetof(ReqWrap<uv_req_t>, req_wrap_queue_)',
+    headers=['req-wrap.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__ReqWrapQueue__headOffset',
+    value='offsetof(Environment::ReqWrapQueue, head_)',
+    headers=['env.h'],
+    type_='size_t',
+  ),
+  DebugSymbol(
+    name='class__ReqWrapQueue__nextOffset',
+    value='offsetof(ListNode<ReqWrap<uv_req_t>>, next_)',
+    headers=['req-wrap.h', 'util.h'],
+    type_='size_t',
+  ),
+]
+
+
+template = '''
+/*
+ * This file is generated by {filename}.  Do not edit directly.
+ */
+
+// #define private public
+// #define protected public
+
+{includes}
+
+{declare_symbols}
+
+namespace node {{
+
+int genDebugSymbols() {{
+{fill_symbols}
+return 1;
+}}
+
+int debugSymbolsGenerated = genDebugSymbols();
```

Comment:
Naming: `int debug_symbols_generated = GenDebugSymbol();`

As well, is it necessary to do it dynamically?  Doesn't `size_t nodedbg_foo_bar = offsetof(Foo, bar);` work?